packages/back-end/src/services/experimentNotifications.ts

```diff
@@ -441,37 +411,92 @@ export const notifySignificance = async ({
   );
 };
 
+export const notifyDecision = async ({
+  context,
+  experiment,
+  currentStatus,
+  lastStatus,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  currentStatus: RunningExperimentStatusData;
+  lastStatus?: RunningExperimentStatusData;
+}) => {
+  if (
+    currentStatus.status === "ship-now" ||
+    currentStatus.status === "rollback-now" ||
+    currentStatus.status === "ready-for-review"
+  ) {
+    const eventType: "ship" | "rollback" | "review" = (() => {
+      switch (currentStatus.status) {
+        case "ship-now":
+          return "ship";
+        case "ready-for-review":
+          return "review";
+        case "rollback-now":
+          return "rollback";
+      }
+    })();
+
+    if (currentStatus.status !== lastStatus?.status) {
+      dispatchEvent({
+        context,
+        experiment,
+        event: `decision.${eventType}`,
+        data: {
+          object: {
+            experimentId: experiment.id,
+            experimentName: experiment.name,
+            decisionDescription: currentStatus.tooltip,
+          },
+        },
+      });
+    }
+  }
+};
+
 export const notifyExperimentChange = async ({
   context,
+  experiment,
   snapshot,
+  lastAnalysisSummary,
 }: {
   context: Context;
+  experiment: ExperimentInterface;
   snapshot: ExperimentSnapshotDocument;
+  lastAnalysisSummary?: ExperimentAnalysisSummary;
 }) => {
-  const experiment = await getExperimentById(context, snapshot.experiment);
-  if (!experiment) throw new Error("Error while fetching experiment!");
-
-  // do not fire significance or error events for exploratory analyses
-  if (snapshot.type === "exploratory") return;
-  // do not fire significance or error events for reports
-  if (snapshot.type === "report") return;
-  // do not fire significance events for old snapshots that have no type
-  if (snapshot.type === undefined) return;
-  // do not fire for snapshots where statistics are manually entered in the UI
-  if (snapshot.manual) return;
-
   await notifySignificance({ context, experiment, snapshot });
 
-  const results = getDefaultAnalysisResults(snapshot);
+  const healthSettings = getHealthSettings(
+    context.org.settings,
+    [...accountFeatures[getEffectiveAccountPlan(context.org)]].includes(
```

Comment:
question: can we use `orgHasPremiumFeature` here?

Reply:
Oh wow yeah thank you I was only seeing the FE hooks and forgot about that method.